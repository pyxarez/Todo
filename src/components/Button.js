import React from 'react';

import './Button.res/style.css'

Button.propTypes = {
  type: React.PropTypes.string,
  onClick: React.PropTypes.func,
}

export default function Button({type, onClick, value}) {
  return type
    ? <button
        className={`my-button-component button-${type}`}
        onClick={onClick}>{value}</button>
    : <button
        className={'my-button-component'}
        onClick={onClick}>{value}</button>
}
