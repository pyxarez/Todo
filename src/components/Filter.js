import React, { Component } from 'react';
import { browserHistory } from 'react-router';

import Button from './Button';
import './Filter.res/style.css';

export default class Filter extends Component {
  static propTypes = {
    URLParams: React.PropTypes.shape({
      id: React.PropTypes.number,
      category: React.PropTypes.string,
      filter: React.PropTypes.string
    })
  }

  handleClearClick = () => {
    this.textInput.value = '';

    const {
      id,
      category
    } = this.props.URLParams;

    if ((id !== undefined) && category) {
      const path = `/main/${id}/${category}/`;
      browserHistory.push(path);
    }
  }

  handleInputUpdate = (e) => {
    const {
      id,
      category
    } = this.props.URLParams;

    if ((id !== undefined) && category) {
      const path = `/main/${id}/${category}/${e.target.value}`;
      browserHistory.push(path);
    }
  }

  render() {
    return (
      <div className="my-filter-component">
        <input
          ref={(input) => { this.textInput = input }}
          onChange={this.handleInputUpdate}
          className="filter__input"
          type="text"
          placeholder="Search"/>
        <Button onClick={this.handleClearClick} type="clear"/>
      </div>
    );
  }
}