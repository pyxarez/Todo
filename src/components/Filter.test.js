import React from 'react';
import { mount, shallow } from 'enzyme';

import Filter from './Filter';
import Button from './Button';

jest.mock('react-router', () => {
  return {
    browserHistory: {
      push: jest.fn()
    }
  };
});
import { browserHistory } from 'react-router';

const setup = () => {
  const props = {
    URLParams: {
      id: 0,
      category: 'Films',
      filter: '54312'
    }
  };

  const wrapper = mount(
    <Filter {...props}/>
  );

  return {
    props,
    wrapper
  }
};

describe('Component :: Filter', () => {
  afterEach(() => {
    jest.resetAllMocks()
  });
  
  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should go to corresponding URL when input update', () => {
    const { wrapper, props } = setup();

    const input = wrapper.find('input');
    const value = 'i"m value';
    input.prop('onChange')({
      target: {
        value
      }
    });

    expect(browserHistory.push)
      .toBeCalledWith(`/main/${props.URLParams.id}/${props.URLParams.category}/${value}`);
  });

  it('should go to corresponding URL when input is cleared', () => {
    const { wrapper, props } = setup();

    const buttonComponent = wrapper.find(Button);
    buttonComponent.simulate('click');
    
    expect(browserHistory.push)
      .toBeCalledWith(`/main/${props.URLParams.id}/${props.URLParams.category}/`);
  });
});