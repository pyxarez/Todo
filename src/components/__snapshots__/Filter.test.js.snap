exports[`Component :: Filter should render self and subcomponents 1`] = `
<Filter
  URLParams={
    Object {
      "category": "Films",
      "filter": "54312",
      "id": 0,
    }
  }>
  <div
    className="my-filter-component">
    <input
      className="filter__input"
      onChange={[Function]}
      placeholder="Search"
      type="text" />
    <Button
      onClick={[Function]}
      type="clear">
      <button
        className="my-button-component button-clear"
        onClick={[Function]} />
    </Button>
  </div>
</Filter>
`;
